master/buildbot/scripts/upgrade_master.py

```diff
@@ -106,34 +107,38 @@ def upgradeDatabase(config, master_cfg):
 
 
 @in_reactor
-@defer.inlineCallbacks
 def upgradeMaster(config, _noMonkey=False):
     if not _noMonkey:  # pragma: no cover
         monkeypatches.patch_all()
 
     if not base.checkBasedir(config):
-        defer.returnValue(1)
-        return
+        return defer.succeed(1)
```

Comment:
Isn't 

```
defer.returnValue(1)
return
```

will lead to `upgradeMaster()` deferred to return `1` and  your version:

```
return defer.succeed(1)
```

will lead to waiting of completed deferred and to return `None`?